server/app.py

```diff
@@ -104,38 +181,105 @@ async def submit(request: SubmitRequest):
     serialized_agent = pickle.dumps(request.agent)
-    logger.debug(f"Submitting task {task.name} with agent {serialized_agent}")
+
+    logger.info(f"Submitting task {task.name} with agent {serialized_agent}")
     result = task.delay(serialized_agent=serialized_agent)
-    logger.debug(f"Task {task.name} submitted with job_id {result.id}")
+    logger.info(f"Task {task.name} submitted with job_id {result.id}")
+
     return Response[JobCreated](data=JobCreated(job_id=result.id))
 
 
-class Status(Enum):
-    PENDING = "Pending"
-    INPROGRESS = "InProgress"
-    COMPLETED = "Completed"
-    FAILED = "Failed"
-    CANCELED = "Canceled"
+async def poll_for_streaming_results(job_id: str, batch_size: int, handler: Callable[List[dict], None]):
+    """
+    Poll for results in the kafka output topic and handle them.
+    """
+    logger.info(f"Polling for results {job_id=}")
+
+    consumer = AIOKafkaConsumer(
+        f"adala-output-{job_id}",
+        bootstrap_servers=settings.kafka_bootstrap_servers,
+        value_deserializer=lambda v: json.loads(v.decode("utf-8")),
+        auto_offset_reset="earliest",
+        # group_id="adala-consumer-group",
+    )
+    await consumer.start()
+    logger.info(f"consumer started {job_id=}")
 
+    try:
+        data = await consumer.getmany(timeout_ms=30_000, max_records=batch_size)
+        logger.info(f"got batch {data=}")
+        for tp, messages in data.items():
+            assert tp.topic == f"adala-output-{job_id}", f"Unexpected topic: {tp.topic}"
+            if messages:
+                handler(messages)
+            else:
+                logger.info(f"No messages in topic {tp.topic}")
+            if len(messages) > batch_size:
+                # assume we're done
+                logger.info(f"End of stream in topic {tp.topic}")
+                break
+    finally:
+        await consumer.stop()
+
+
+@app.post("/jobs/submit-streaming", response_model=Response[JobCreated])
+async def submit_streaming(request: SubmitStreamingRequest, background_tasks: fastapi.BackgroundTasks):
+    """
+    Submit a request to execute task `request.task_name` in celery.
 
-class JobStatusResponse(BaseModel):
+    Args:
+        request (SubmitStreamingRequest): The request model for submitting a job.
+
+    Returns:
+        Response[JobCreated]: The response model for a job created.
     """
-    Response model for getting the status of a job.
 
-    Attributes:
-        status (str): The status of the job.
-        processed_total (List[int]): The total number of processed records and the total number of records in job.
-            Example: [10, 100] means 10% of the completeness.
+    # TODO: get task by name, e.g. request.task_name
+    task = process_file_streaming
+    serialized_agent = pickle.dumps(request.agent)
+
+    logger.info(f"Submitting task {task.name} with agent {serialized_agent}")
+    result = task.delay(serialized_agent=serialized_agent)
+    # TODO pass batch size in request?
```

Comment:
imo we could maybe make batch size an optional param, but I don't see too much value there atm (could be missing something)